models/newsletter.js

```diff
@@ -0,0 +1,78 @@
+const { loggers } = require("winston");
+const firestore = require("../utils/firestore");
+const newsletterModel = firestore.collection("newsletter");
+
+const DB_ERROR = "Not able to conenct to DB";
+
+const subscribe = async (data) => {
+  try {
+    const { email } = data;
+    const check = await newsletterModel.where("email", "==", email).get();
+    let exist = false;
+    check.forEach((ch) => {
+      const data = ch.data();
+      if (!data.active) {
+        const res = ch.id;
+        newsletterModel.doc(res).update({
+          active: true,
+          email: data.email,
+        });
+      }
+      exist = true;
+    });
+    if (!exist) {
+      await newsletterModel.add({
+        email: email,
+        active: true,
+      });
+      return "new added";
+    }
+    return "already added in the list";
```

Comment:
This can be something like `Email already present in the mailing list`

Reply:
done